### PR TITLE
Add a metrics for cdc to observe the region count that use too long to do incremental scan

### PR DESCRIPTION
**Pull request by hongyunyan**
### What is changed and how it works?

Issue Number: ref https://github.com/tikv/tikv/issues/12592

What's Changed:
Add a new metrics called tikv_cdc_scan_long_duration_region to observe the region count that use too long to do incremental scan.

```commit-message

```

### Related changes

- [ ] PR to update `pingcap/docs`/`pingcap/docs-cn`:
- [ ] Need to cherry-pick to the release branch

### Check List

Tests 

- [ ] Unit test
- [ ] Integration test
- [ ] Manual test (add detailed scripts or steps below)
- [ ] No code

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

### Release note

```release-note
None
```

### CODE REVIEW

File: components/cdc/src/initializer.rs [hunk 1]
```diff
@@ -266,7 +266,14 @@ impl<E: KvEngine> Initializer<E> {
             DownstreamState::Initializing | DownstreamState::Stopped
         ));
 
+        let mut scan_long_time = false;
         while !done {
+            // Add metrics to observe long time incremental scan region count
+            if (!scan_long_time) && start.saturating_elapsed() > Duration::from_secs(60) {
+                CDC_SCAN_LONG_DURATION_REGIONS.inc();
+                scan_long_time = true;
+                warn!("cdc incremental scan takes too long"; "region_id" => region_id, "conn_id" => ?self.conn_id, "downstream_id" => ?self.downstream_id, "takes" => ?start.saturating_elapsed());
```
Comment (hicqu) [inline components/cdc/src/initializer.rs:275]: The line is too long.

File: components/cdc/src/initializer.rs [hunk 2]
```diff
@@ -266,7 +266,14 @@ impl<E: KvEngine> Initializer<E> {
             DownstreamState::Initializing | DownstreamState::Stopped
         ));
 
+        let mut scan_long_time = false;
         while !done {
+            // Add metrics to observe long time incremental scan region count
```
Comment (hicqu) [inline components/cdc/src/initializer.rs:271]: The function can return in the `while` loop. So please consider to use `defer` to decrease the metric.

**Review by hicqu — APPROVED** [review]

File: components/cdc/src/initializer.rs [hunk 3]
```diff
@@ -266,7 +266,18 @@ impl<E: KvEngine> Initializer<E> {
             DownstreamState::Initializing | DownstreamState::Stopped
         ));
 
+        let mut scan_long_time = false;
+
         while !done {
+            // Add metrics to observe long time incremental scan region count
+            if (!scan_long_time) && start.saturating_elapsed() > Duration::from_secs(60) {
+                CDC_SCAN_LONG_DURATION_REGIONS.inc();
+                defer!(CDC_SCAN_LONG_DURATION_REGIONS.dec());
```
Comment (YuJuncen) [inline components/cdc/src/initializer.rs:275]: Will this `defer` execute right after the if block exits?

**Review by YuJuncen — APPROVED** [review]
Rest LGTM

**Review by YuJuncen — APPROVED** [review]

**Pull request merged**